WordPress/Classes/ViewRelated/Post/PostSettingsViewController+FeaturedImageUpload.swift

```diff
@@ -73,12 +73,12 @@ extension PostSettingsViewController {
         guard let media = MediaCoordinator.shared.addMedia(from: asset, to: apost) else {
             return
         }
-        self.apost.featuredImage = media
+        self.apost.featuredImage = try? self.apost.managedObjectContext?.existingObject(with: TaggedManagedObjectID(media))
```

Comment:
This is what I referred to in the description. When you add a `featuredImage`, it gets added to `viewContext`, so we have to get the instance in the temporary child context. The child context also automatically merges the changes from `viewContext` to allow the screen to show the updated status of the `Media` asset. This is probably the only complication with child contexts.